pkg/network/tracer/ebpf_conntracker.go

```diff
@@ -113,88 +115,47 @@ func NewEBPFConntracker(cfg *config.Config) (netlink.Conntracker, error) {
 
 func (e *ebpfConntracker) dumpInitialTables(ctx context.Context, cfg *config.Config) error {
 	e.consumer = netlink.NewConsumer(cfg.ProcRoot, cfg.ConntrackRateLimit, true)
-	e.decoder = netlink.NewDecoder()
 	defer e.consumer.Stop()
 
 	for _, family := range []uint8{unix.AF_INET, unix.AF_INET6} {
-		events, err := e.consumer.DumpTable(family)
+		done, err := e.consumer.DumpAndDiscardTable(family)
 		if err != nil {
 			return err
 		}
-		if err := e.loadInitialState(ctx, events); err != nil {
+
+		if err := e.processEvents(ctx, done); err != nil {
 			return err
 		}
 	}
+
+	go func() {
+		// We'll wait a while before detaching the hook in order to give ebpf ample time to finish processing the dump
+		timer := time.NewTimer(1 * time.Minute)
```

Comment:
What is the reason for a timer? Are we not confident in getting a stop message on the channel?

Reply:
... good point, I'm not sure why I thought the dump was being processed asynchronously when I wrote this. We do wait to have received (and dropped) all the data from the table before returning.